### PR TITLE
Linking to an expanded workshop archive

### LINKED ISSUE #167: QoL Change: Linking to an expanded workshop archive
Hello hello! Always a huge fan of the Hack website, and in particular, I wanted to share with my devs some of the previous resources y'all have made. This is a very small QoL change, but is it possible to link to the `/archive/` page with a category already expanded?

I'm imagining: I link `https://hack.uclaacm.com/archive?2019-hackschool` or `https://hack.uclaacm.com/archive/#2019-hackschool`

And the user loads in to:

Screen Shot 2021-02-18 at 7 54 18 PM

Normally, I just tell them to scroll down to a header and click to expand (which is totally fine), but I think this might be a cool addition! Any thoughts? Not sure how this plays into Gatsby's router!

p.s. still sharing Hackschool and JS Chats content 😊

### PR DESCRIPTION
**Pull request by katelynsyu**
# Changes

- add ID to all accordions, so that adding quarter-year-eventName to the archive url will take you to the relevant accordion
- add a button that will copy the link to the accordion
- when linked to the accordion, the accordion will load as expanded

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)

# Related Issues

Fixes #167

### CODE REVIEW

**Comment by katelynsyu** [comment]
to do:
- [ ] ensure that clicking the link icon doesn't open the accordion
- [ ] if the url is to a valid accordion, set isExpanded to true and render it expanded.

File: src/components/ArchivePage/EventInfoItem.js [hunk 1]
```diff
@@ -50,28 +53,54 @@ const useStyles = makeStyles(theme => ({
...
 
-function EventInfoItem({ name, mainLink, tags, directors, workshops, tagHighlight }) {
+function EventInfoItem({ name, mainLink, tags, directors, workshops, tagHighlight, slug }) {
 	const classes = useStyles();
-	const [isExpanded, setExpanded] = useState(true);
-	return <Accordion classes={{ root: classes.paperRoot }}
-		onChange={() => setExpanded(e => !e)}>
+	const [isExpanded, setExpanded] = useState(false);
+	console.log(name)
+	console.log(`window: `, window.location.hash)
```
Comment (khxia) [inline src/components/ArchivePage/EventInfoItem.js:60]: looks like this line is causing our build to fail. Try to remove it and see if running `yarn build` works.

File: src/components/ArchivePage/EventInfoItem.js [hunk 2]
```diff
@@ -50,28 +53,54 @@ const useStyles = makeStyles(theme => ({
...
+	const [hash, setHash] = useState(window.location.hash);
```
Comment (khxia) [inline src/components/ArchivePage/EventInfoItem.js:61]: I actually think generally we aren't able to use the `window` object in the function body except in a react useEffect. https://blog.greenroots.info/gatsby-the-window-is-not-defined-error-what-and-how-to-fix-it 

Could you try play around with it and see what works in yarn build?

**Review by khxia — CHANGES_REQUESTED** [review]
Our build is failing rip

File: src/components/ArchivePage/ArchivePageTemplate.js [hunk 3]
```diff
@@ -67,19 +67,25 @@ function ArchivePageTemplate({ pageContext }) {
...
-					<EventInfoItem
-						key={event.name}
-						name={event.name}
-						mainLink={event.mainLink}
-						tags={event.tags}
-						directors={event.directors}
-						workshops={event.workshops}
-					/>)}
+				{quarterEvents[quarter].map(event => {
+					// name = quarter + " " + event.name
```
Comment (khxia) [inline src/components/ArchivePage/ArchivePageTemplate.js:75]: get rid of these comments

**Review by khxia — CHANGES_REQUESTED** [review]

**Review by khxia — APPROVED** [review]
Looks good to me!

**Pull request merged**